hack/install.sh

```diff
@@ -20,213 +20,215 @@ command_exists() {
+                set -x
+                if [ "https://get.docker.com/" = "$url" ]; then
+                    $sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 36A1D7869245C8950F966E92D8576A8BA88D21E9"
+                elif [ "https://test.docker.com/" = "$url" ]; then
+                    $sh_c "apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 740B314AE3941731B942C66ADF4FD13717AAD7D6"
+                else
+                    $sh_c "$curl ${url}gpg | apt-key add -"
+                fi
+                $sh_c "echo deb ${url}ubuntu docker main > /etc/apt/sources.list.d/docker.list"
+                $sh_c 'sleep 3; apt-get update; apt-get install -y -q lxc-docker'
+            )
+            if command_exists docker && [ -e /var/run/docker.sock ]; then
+                (
+                    set -x
+                    $sh_c 'docker version'
+                ) || true
+            fi
+            your_user=your-user
+            [ "$user" != 'root' ] && your_user="$user"
+            echo
+            echo 'If you would like to use Docker as a non-root user, you should now consider'
+            echo 'adding your user to the "docker" group with something like:'
+            echo
+            echo '  sudo usermod -aG docker' $your_user
+            echo
+            echo 'Remember that you will have to log out and back in for this to take effect!'
+            echo
+            exit 0
+            ;;
+
+        gentoo)
+            if [ "$url" = "https://test.docker.com/" ]; then
+                echo >&2
+                echo >&2 '  You appear to be trying to install the latest nightly build in Gentoo.'
+                echo >&2 '  The portage tree should contain the latest stable release of Docker, but'
+                echo >&2 '  if you want something more recent, you can always use the live ebuild'
+                echo >&2 '  provided in the "docker" overlay available via layman.  For more'
+                echo >&2 '  instructions, please see the following URL:'
+                echo >&2 '    https://github.com/tianon/docker-overlay#using-this-overlay'
+                echo >&2 '  After adding the "docker" overlay, you should be able to:'
+                echo >&2 '    emerge -av =app-emulation/docker-9999'
+                echo >&2
+                exit 1
+            fi
+
+            (
+                set -x
+                $sh_c 'sleep 3; emerge app-emulation/docker'
+            )
+            exit 0
+            ;;
+    esac
+
+    echo >&2
+    echo >&2 'Either your platform is not easily detectable, is not supported by this'
+    echo >&2 'installer script (yet - PRs welcome! [hack/install.sh]), or does not yet have'
+    echo >&2 'a package for Docker.  Please visit the following URL for more detailed'
+    echo >&2 'installation instructions:'
+    echo >&2
+    echo >&2 ' https://docs.docker.com/en/latest/installation/'
+    echo >&2
+}
+
+do_docker_install
 exit 1
```

Comment:
My thought was that if the function exited for any reason we'd still want to exit 1. I've added an exit inside of the function. It should be impossible to reach the outside one, but I've left it there in case a bug is introduced.